src/server.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const moment = require('moment');
+const cudl = require('cuddle');
+const _ = require('lodash');
+
+const config = require('./config');
+const cache = require('./cache');
+
+module.exports = {
+    verify_scopes
+};
+
+function verify_scopes (required_scopes) {
+    return (req, res, next) => {
+        return check_scopes(req, required_scopes)
+            .then(result => {
+                req.client_id = result.client_id;
+                next();
+            })
+            .catch(error => next(error));
+    };
+}
+
+function check_scopes (request, required_scopes) {
+    const access_token = request.headers['access-token'];
+
+    return check_input(access_token)
+        .then(get_token_info)
+        .then(result => ({access_token, result, required_scopes}))
+        .then(check_token_validity);
+}
+
+function check_input (access_token) {
+    return new Promise((accept, reject) => {
+        if (!access_token) {
+            return reject(new Error('Access token is required.'));
+        }
+
+        accept(access_token);
+    });
+}
+
+function get_token_info (access_token) {
+    const cached_token_info_result = cache.get('server', access_token);
+
+    if (_.has(cached_token_info_result, 'date_expiration')) {
+        const date_expiration = moment.utc(cached_token_info_result.date_expiration, 'x');
+        const token_is_expired = date_expiration.isBefore(moment());
+
+        if (!token_is_expired) {
+            return Promise.resolve(cached_token_info_result);
```

Comment:
just return the value. its already a thenable

Reply:
I think we should return Promise.resolve for consistency since the alternate path returns a promise (return of cudl promise).